src/test/java/org/prebid/server/bidder/vox/VoxBidderTest.java

```diff
@@ -0,0 +1,203 @@
+import org.prebid.server.bidder.model.HttpResponse;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+
+import java.util.List;
+import java.util.function.Function;
+
+import static java.util.function.Function.identity;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.prebid.server.bidder.model.BidderError.badServerResponse;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.audio;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.banner;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.video;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.xNative;
+import static org.prebid.server.util.HttpUtil.ACCEPT_HEADER;
+import static org.prebid.server.util.HttpUtil.APPLICATION_JSON_CONTENT_TYPE;
+import static org.prebid.server.util.HttpUtil.CONTENT_TYPE_HEADER;
+import static org.prebid.server.util.HttpUtil.headers;
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;
+
+class VoxBidderTest extends VertxTest {
+
+    private static final String ENDPOINT_URL = "https://test-url.com";
+    private static final String CURRENCY = "USD";
+
+    private VoxBidder sut;
+
+    @Test
+    public void shouldFailOnBidderCreation() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new VoxBidder("invalid_url", jacksonMapper));
+    }
+
+    @Test //todo: not sure whether it's fine
+    public void shouldMakeCorrectRequestWhenRequestDoesNotHaveImpressions() {
+        // given
+        final var bidRequest = givenBidRequest();
+        // when
+        final var result = sut.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).usingRecursiveComparison().isEqualTo(expectedResults(bidRequest).getValue());
+    }
+
+    @Test
+    public void shouldMakeCorrectRequest() {
+        // given
+        final var bidRequest = givenBidRequest(identity());
+
+        // when
+        final var result = sut.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).usingRecursiveComparison().isEqualTo(expectedResults(bidRequest).getValue());
+    }
+
+    @Test
+    public void shouldReturnErrorWhileMakingBidsWhenResponseCanNotBeParsed() {
+        // given
+        final var httpCall = givenHttpCall(null, "invalid");
+
+        // when
+        final var actual = sut.makeBids(httpCall, null);
+
+        // then
+        assertThat(actual.getValue()).isEmpty();
+        assertThat(actual.getErrors()).containsExactly(badServerResponse("Bad Server Response"));
+    }
+
+    @Test
+    public void shouldReturnBids() {
+        // given
+        final var bannerBid = Bid.builder().impid("1").mtype(1).build();
```

Comment:
We are not using var, only strict types

Reply:
var looks neat, I mean we can use them at least in tests

Reply:
For some historical reasons, we agreed to use only strict types.

Also test code style should be similar to prod code style, easier for developer to create new code